### PR TITLE
Fix/protonation

### PR DESCRIPTION
**Pull request by fgrunewald**
This PR fixes most perhaps all issues we have regarding neutral amino-acids and pronation states. It implements neutral protonated ASP and GLU as ASPP and GLUP as well as neutral LYS as LYN. Furthermore, it takes care of implementing the three HIS variants based on the naming scheme used in CHARMM and Amber. To that end the universal force-field is abandoned in favor of charmm. It always was charmm based anyways. In addition the amber force-field is implemented which allows using naming from the amber force-field family. This should cover most of the HIS naming schemes. Mapping PDBs directly from the PDB will still work the same way as previously with universal because the HIS residue still maps to HSD/HID respectively for the two force-field families. Abandoning universal has the advantage that usage should be more clear now. That is if someone maps an amber based PDB with names HIE/HID/HIP there will be the warning that these residues cannot be found. But simply changing the default `-from charmm` to `-from amber` solves the issue. It also makes clear that we don't support any other naming variants unlike universal that somehow suggested we could magically pick this up.  

To Do 
---------

- [x] implement HIS Charmm
- [x] implement HIS Amber
- [x] check Amber names for neutral LYS, GLU, ASP
- [x] check that modifications would in principle work
- [x] add integration tests
- [x] make sure to add amino-acid names to all hidden variable definitions 
~~- [ ] add capped termini~~

Fixes
-------
#381 #323 #376 #417 #412 #395 #419

### CODE REVIEW

File: bin/martinize2
Comment (pckroon) [inline bin/martinize2:107]: Should this be charmm36?

**Review by pckroon — COMMENTED** [review]
Fantastic :heart: 
I'll take a more detailed look at the data files soon (TM). I do think all FFs should be versioned, so not charmm but charmm36. Ditto for Amber.

Great work.
Will this make v1.0.0?

File: bin/martinize2
Comment (fgrunewald) [reply bin/martinize2:107]: I don't think we need the version identifier, because all amber family or charmm family pdbs can be mapped to martini without issues. That is independent of the actual version. So charmm27 will work as well as charmm36 or charmm36m. In fact that is what universal already did to some extent. Naming the ff versions will add a huge amount of redundant files, that we don't really need. It would only be important in case we would want to generate AA parameters, but we already dropped this from the scope. So keeping it just charmm will be fine.

Comment (pckroon) [reply bin/martinize2:107]: Ok, agreed. We can always change it at a later stage (together with a smarter file format to reduce some redundancy)

**Comment by pckroon** [comment]
Please rebase this PR on master and force push the result to resolve the merge conflicts

**Review by pckroon — CHANGES_REQUESTED** [review]
I'm fine with pushing termini to a separate PR. It's indeed good to keep things small-ish.

File: vermouth/data/force_fields/amber/aminoacids.rtp
Comment (pckroon) [inline vermouth/data/force_fields/amber/aminoacids.rtp:86]: Do we want to keep water and ions? Same question goes for everything that's not an amino acid (URE, ACE, ...)

File: vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command
Comment (fgrunewald) [reply vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command:11]: because there is one hydrogen in the crystal structure that is within the bond cut-off of a neighboring hydrogen thus forming a triangle.

Comment (fgrunewald) [reply vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command:11]: at lest now the maxwarn is tested ^^

File: vermouth/data/force_fields/amber/aminoacids.rtp
Comment (fgrunewald) [reply vermouth/data/force_fields/amber/aminoacids.rtp:86]: I'll remove that stuff. I didn't think about it when copying the GMX file to be honest

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (pckroon) [inline vermouth/data/force_fields/martini3001/aminoacids.ff:638]: intended?

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (pckroon) [inline vermouth/data/force_fields/martini3001/aminoacids.ff:645]: The bond has distance 0.32, the constraint 0.30?

File: vermouth/data/force_fields/martini3001/modifications.ff
Comment (pckroon) [inline vermouth/data/force_fields/martini3001/modifications.ff:50]: This looks a little fishy. Is it a leftover test case?

Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/modifications.ff:50]: if you don't have the edges an error is raised from the connected graph check for modifications

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:645]: @paulocts ?

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:516]: @paulocts?

File: vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command
Comment (pckroon) [reply vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command:11]: Maxwarn is also unittested. 
Is it an option to remove that H from the aa pdb? I'm worried that at some later point the warning goes away (for whatever reason), and the maxwarn starts masking another, harmful, warning.

File: vermouth/tests/integration_tests/test_integration.py
Comment (pckroon) [inline vermouth/tests/integration_tests/test_integration.py:133]: What's the difference between lysosyme_prot and lysozyme? Could you put that in the README of the _prot testcase?

**Review by pckroon — COMMENTED** [review]
`vermouth/data/mappings/martini3001/.modifications.mapping.swo` should not be committed! Please remove and rebase so it's also not in the history. Also update `.gitignore` ;)
`vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/screeen.output` -> `screen.output`?

File: vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command
Comment (fgrunewald) [reply vermouth/tests/data/integration_tests/tier-1/1mj5/martinize2/command:11]: ok I'll remove the H

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:242]: isn't it either or?

File: vermouth/tests/integration_tests/test_integration.py
Comment (fgrunewald) [reply vermouth/tests/integration_tests/test_integration.py:133]: added README; FYI this one tests the different protonation states for the charmm AAs

File: vermouth/data/force_fields/martini3001/modifications.ff
Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/modifications.ff:50]: let's keep it for now. we need to change these things later anyways. but this modification can be a paper example

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (pckroon) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:242]: Yes. What you do here duplicates a lot of data (bonds, angles, whatnots). In the Ideas and Opinions I have about the data files this should not be necessary, but it depends on working resname aliases (e.g. ASPP = ASP + ASP-prot). So this is a TODO for me, for later.

**Review by pckroon — APPROVED** [review]
LGTM, if you're sure of the HIS parameters.

**Comment by fgrunewald** [comment]
@pckroon I'm waiting for confirmation of the parameters but that could take a while

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (paulocts) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:645]: For HSP, both should be 0.32.

Comment (paulocts) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:645]: I mean, both distances SC1-SC2 and SC1-SC3... and both constraint and the bond in the flexible option. All 0.32 nm

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (paulocts) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:516]: In my tests, I never used this improper dihedral for any HIS or PHE .... and it never caused me any instability problem. As It is commented and many users are using in this way, I would remove these lines or continue to keep commented as it is. We will come back to this point in the future for protein model.

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (paulocts) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:638]: As mentioned in the answer below. For any protonated HIS models, SC1-SC2 and SC1-SC3 bond lengths should be 0.32 nm.

File: vermouth/data/force_fields/martini3001/aminoacids.ff
Comment (fgrunewald) [reply vermouth/data/force_fields/martini3001/aminoacids.ff:516]: ok I removed those

**Comment by fgrunewald** [comment]
@pckroon this looks some hypothesis stuff happening here?!

**Review by pckroon — DISMISSED** [review]

**Comment by pckroon** [comment]
Jup. In `vermouth/tests/gmx/test_gro.py:410` `hnp.scalar_dtypes()` should probably be changed to e.g. `st.choice(hnp.integer_dtypes, hnp.floating_dtypes, hnp.complex_number_dtypes)`, because well, we don't care about dates. Quicker fix might be to add an appropriate `assume` afterwards forbidding datetime objects. I'm ok with rerunning tests until hypothesis is happy though.

**Review by pckroon — APPROVED** [review]
LGTM, nice work!

**Pull request merged**
